tuf/api/metadata.py

```diff
@@ -0,0 +1,148 @@
+# 1st-party.
+from keys import (
+    Keyring,
+    Threshold,
+    get_private_keys_from_keyring,
+    get_public_keys_from_keyring,
+    sorted_list_of_keyids,
+    write_and_read_new_keys,
+)
+
+# 2nd-party.
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+import json
+
+# 3rd-party.
+from dateutil.relativedelta import relativedelta
+from securesystemslib.formats import encode_canonical
+from securesystemslib.keys import create_signature, verify_signature
+from securesystemslib.util import load_json_file
+from tuf.repository_lib import (
+    _get_written_metadata,
+    generate_snapshot_metadata,
+    generate_targets_metadata,
+    generate_timestamp_metadata,
+)
+
+# Types.
+JsonDict = Dict[str, Any]
+
+# Classes.
+
+class Metadata:
+    # By default, a Metadata would be a rather empty one.
+    def __init__(self, consistent_snapshot: bool = True, expiration: relativedelta = relativedelta(), keyring: Optional[Keyring] = None, version: int = 1) -> None:
+        self.consistent_snapshot = consistent_snapshot
+
+        self.keyring = keyring
+        self.expiration = expiration
+
+        assert version > 1, f'{version} < 1'
+        self.version = version
+
+    # And you would use this method to populate it from a file.
+    def read_from_json(self, filename: str) -> None:
+        signable = load_json_file(filename)
+
+        # TODO: use some basic schema checks
+        signatures = signable['signatures']
+        signed = signable['signed']
+
+        self.expiration = datetime.strptime(signed['expiration'], '%b %d %Y %I:%M%p')
+        self.version = signed['version']
```

Comment:
TODO: should be able to set appropriate values for `self.consistent_snapshot` at load, also.